composer.json

```diff
@@ -1,159 +1,139 @@
-            },
-            "drupal\/fakeobjects": {
-                "3002953: Fix js plugin seems to be missing": "https:\/\/www.drupal.org\/files\/issues\/2020-06-22\/fakeobjects-missing_plugin-3002953-5.patch"
-            },
-            "drupal\/core": {
-                "#2951547: Fix issue with layout overflow": "https:\/\/www.drupal.org\/files\/issues\/2020-01-08\/layout-builder-tray-size-2951547-42.patch",
-                "3102249: Changing an existing embedded media's attributes does not get saved with CKEditor": "https:\/\/www.drupal.org\/files\/issues\/2020-06-01\/3102249-21.patch",
-                "3109650: Refactor Xss::attributes() to allow filtering of style attribute values": "https:\/\/www.drupal.org\/files\/issues\/2021-09-22\/core-9.2-xss-refactor_filter_attributes-3109650-43.patch",
-                "3097338: Fix issue with image replacement in ckeditor": "https:\/\/www.drupal.org\/files\/issues\/2021-06-03\/drupal-3097338-68.patch"
-            },
-            "drupal\/config_readonly": {
-                "#3085001: Config blacklist to only block some configuration while in read only mode": "https:\/\/www.drupal.org\/files\/issues\/2019-10-31\/config_readonly-3085001-2.patch"
-            },
-            "drupal\/field_states_ui": {
-                "#3225009: Cant add custom block with field states ui using layout builder": "https:\/\/www.drupal.org\/files\/issues\/2022-03-07\/3225009-5.patch"
-            },
-            "drupal\/linkit": {
-                "2877535: Fix issue with path alias": "https:\/\/www.drupal.org\/files\/issues\/2022-06-07\/2877535-39.patch"
-            },
-            "drupal\/allowed_formats": {
-                "3269869: Drupal 10 compatibility": "https:\/\/git.drupalcode.org\/project\/allowed_formats\/-\/commit\/9b52e52.patch"
-            },
-            "drupal\/fontawesome": {
-                "#3219579: Hidding font awesome additional settings": "https:\/\/www.drupal.org\/files\/issues\/2022-12-14\/hidding_font_awesome_additional_settings-3219579-18.patch"
-            },
-            "drupal\/cas": {
-                "3318382: TypeError: in_array(): Argument #2 ($haystack) must be of type array": "https:\/\/www.drupal.org\/files\/issues\/2022-10-31\/cas-null-in_array-3318382-0.patch"
-            },
-            "drupal\/ckeditor": {
-                "#3095304: Changes to a block's text field not saved when ckeditor source mode is active": "https:\/\/www.drupal.org\/files\/issues\/2023-01-05\/changes_to_a_blocks_text_field_not_saved_3095304-47_3.patch"
-            },
-            "drupal\/paragraphs": {
-                "#3090200: Paragraphs do not render: access check for view fail when using layout builder": "https:\/\/www.drupal.org\/files\/issues\/2022-10-26\/3084934-13_combine_3090200-22.patch"
-            }
+        "patches-file": "composer.patches.json",
```

Comment:
Curious: Can `patches` and `patches-file` be used at the same time - enabling us to make a recommendation that site owners use one approach and we use the other, so we avoid conflicts?

Reply:
Or alternately, can patches be included in the `custom-dependencies/composer.json` ? I'm thinking the answer is no.  Would be good to highlight that in the `custom-dependencies/README.md`.

Reply:
@mlsamuelson I had tried moving the patches to the `custom-dependencies` folder, but they were always skipped. It seems they need to be in the root composer file at the end of the day (whether loaded into it or place directly into it) based on the order in which composer operates.

As for using the patches and the patches file, I have not yet tried that, so only one way to find out. My gut says it would conflict since there would be two `patches` keys in use, but we will see.

Reply:
Just tested it out, moved one patch out of the patches file and directly into the composer file. Even though the patches file was declared before the patch itself, only the patch in the composer file was applied and all patches in the patches file were ignored.

<img width="986" alt="Screenshot 2023-05-09 at 4 02 48 PM" src="https://github.com/ASUWebPlatforms/webspark-ci/assets/5995907/34d88cd8-1d77-49f8-8145-da040598f4f7">

We would need to work out @tbutterf idea for full conflict avoidance.

Reply:
Keep in mind, that we as upstream maintainers are not supposed to be editing the root composer file at all according to Pantheon, so I think we should always try to frame our workflow in that perspective.

Reply:
@mlsamuelson and @bebaps, I have gotten my idea to work. I created a branch and a pull request based on your branch so you can see what I did. It seems to test out initially, but will need some further testing and possible refining. https://github.com/ASUWebPlatforms/webspark-ci/pull/14

Reply:
@tbutterf as per the group Zoom today, we were able to test your PR and confirm it worked as expected. We also tested if the same patch was applied in both files, that the subsequent generated `composer.patches.json` file only included the patch once, and the current method of combining the 2 patch files was adequate as is.